File: app/src/main/java/com/example/simplelibrary/data/entity/Book.java
package com.example.simplelibrary.data.entity;

import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "books", indices = {@Index(value = {"title"}, unique = true)})
public class Book {

    @PrimaryKey(autoGenerate = true)
    public long id;

    public String title;
    public String author;
    public int year;
    public int totalCopies;
    public int availableCopies;

    public Book(String title, String author, int year, int totalCopies, int availableCopies) {
        this.title = title;
        this.author = author;
        this.year = year;
        this.totalCopies = totalCopies;
        this.availableCopies = availableCopies;
    }
}

File: lib/models/book.dart
class Book {
  final int id;
  final String title;
  final String author;
  final int stock;

  Book({required this.id, required this.title, required this.author, required this.stock});

  factory Book.fromMap(Map<String, Object?> m) => Book(
        id: (m['id'] as int),
        title: (m['title'] as String),
        author: (m['author'] as String),
        stock: (m['stock'] as int),
      );
}
